feature_engine/variable_manipulation.py

```diff
@@ -119,15 +133,18 @@ def _find_or_check_categorical_variables(
 
     if variables is None:
         # find categorical variables in dataset
-        variables = list(X.select_dtypes(include=["O", "category"]).columns)
+        variables = [
+            column for column in X.select_dtypes(exclude="number").columns
+            if contains_categorical(X[column])
+        ]
         if len(variables) == 0:
             raise ValueError(
                 "No categorical variables found in this dataframe. Please check "
                 "variable format with pandas dtypes."
             )
 
     elif isinstance(variables, (str, int)):
-        if is_categorical(X[variables]) or is_object(X[variables]):
+        if contains_categorical(X[variables]):
```

Comment:
as per previous comment, I would not change the logic then variables is a str, or a list.